x-pack/plugins/ml/public/components/anomalies_table/links_menu.js

```diff
@@ -0,0 +1,407 @@
+            `Unable to open link as an error occurred loading details on category ID ${categoryId}`);
+        });
+
+    } else {
+      // Replace any tokens in the configured url_value with values from the source record,
+      // and then open link in a new tab/window.
+      const urlPath = getUrlForRecord(customUrl, record);
+      window.open(urlPath, '_blank');
+    }
+
+  };
+
+  viewSeries = () => {
+    const record = this.props.anomaly.source;
+    const bounds = this.props.timefilter.getActiveBounds();
+    const from = bounds.min.toISOString();    // e.g. 2016-02-08T16:00:00.000Z
+    const to = bounds.max.toISOString();
+
+    // Zoom to show 50 buckets either side of the record.
+    const recordTime = moment(record.timestamp);
+    const zoomFrom = recordTime.subtract(50 * record.bucket_span, 's').toISOString();
+    const zoomTo = recordTime.add(100 * record.bucket_span, 's').toISOString();
+
+    // Extract the by, over and partition fields for the record.
+    const entityCondition = {};
+
+    if (_.has(record, 'partition_field_value')) {
+      entityCondition[record.partition_field_name] = record.partition_field_value;
+    }
+
+    if (_.has(record, 'over_field_value')) {
+      entityCondition[record.over_field_name] = record.over_field_value;
+    }
+
+    if (_.has(record, 'by_field_value')) {
+      // Note that analyses with by and over fields, will have a top-level by_field_name,
+      // but the by_field_value(s) will be in the nested causes array.
+      // TODO - drilldown from cause in expanded row only?
+      entityCondition[record.by_field_name] = record.by_field_value;
+    }
+
+    // Use rison to build the URL .
+    const _g = rison.encode({
+      ml: {
+        jobIds: [record.job_id]
+      },
+      refreshInterval: {
+        display: 'Off',
+        pause: false,
+        value: 0
+      },
+      time: {
+        from: from,
+        to: to,
+        mode: 'absolute'
+      }
+    });
+
+    const _a = rison.encode({
+      mlTimeSeriesExplorer: {
+        zoom: {
+          from: zoomFrom,
+          to: zoomTo
+        },
+        detectorIndex: record.detector_index,
+        entities: entityCondition,
+      },
+      filters: [],
+      query: {
+        query_string: {
+          analyze_wildcard: true,
+          query: '*'
+        }
+      }
+    });
+
+    // Need to encode the _a parameter in case any entities contain unsafe characters such as '+'.
+    let path = chrome.getBasePath();
+    path += '/app/ml#/timeseriesexplorer';
+    path += '?_g=' + _g;
+    path += '&_a=' + encodeURIComponent(_a);
+    window.open(path, '_blank');
+  }
+
+  viewExamples = () => {
+    const categoryId = this.props.anomaly.entityValue;
+    const record = this.props.anomaly.source;
+    const indexPatterns = getIndexPatterns();
+
+    const job = mlJobService.getJob(this.props.anomaly.jobId);
+    const categorizationFieldName = job.analysis_config.categorization_field_name;
```

Comment:
there is a low risk here that `mlJobService.getJob` will return `undefined` which will cause this line to throw an error